docs/layer_util.js

```diff
@@ -17,56 +18,88 @@ export {
   toggleLayerOn,
 };
 // @VisibleForTesting
-export {layerArray, layerMap, LayerMapValue};
+export {deckGlArray, DisplayedLayerData, layerArray};
 
 const scoreLayerName = 'score';
 
 /**
- * Keep a map of layer name -> data, index, display status. Data is lazily
- * generated i.e. pre-known layers that don't display by default will have an
- * entry in this map, but the LayerMapValue will have a null data field until we
- * fetch the data when the user wants to display it.
+ * All map overlay layers. Data is lazily generated: layers that don't display
+ * by default will have an entry in this map, but the DisplayedLayerData will
+ * have a null data/overlay field (depending on whether or not we render it with
+ * deck) until we fetch the data when the user wants to display it. Score layer
+ * gets its own special 'score' index, exploiting Javascript's tolerance for
+ * non-numerical indices on arrays.
  */
-const layerMap = new Map();
+const layerArray = [];
 
 /**
  * deck.gl layers, in the order they should be rendered. Passed to deckGlOverlay
  * in redrawLayers() (after filtering out absent elements).
  *
- * Contains one GeoJsonLayer per LayerMapValue with non-null data attribute
- * (from layerMap), ordered by LayerMapValue.index.
+ * Contains one GeoJsonLayer per DisplayedLayerData with non-null data attribute
+ * (from layerArray), ordered by DisplayedLayerData.index. Same trick as with
+ * layerArray for score index.
  *
  * @type {Array<deck.GeoJsonLayer>}
  */
-const layerArray = [];
+const deckGlArray = [];
 
 /**
  * Container for all deck.gl layers. Initialized lazily so tests can load file
  * without deck being defined yet.
  */
 let deckGlOverlay;
 
-/** Values of layerMap. */
-class LayerMapValue {
+/**
+ * Values of layerArray. In addition to basic data from constructor, will
+ * have a data attribute for deck layers, and an overlay attribute for
+ * image/other layers.
+ */
+class DisplayedLayerData {
```

Comment:
Class name is somewhat confused since the layer is not always displayed right?

Reply:
Yeah, I'm trying to say that this is data about how a layer is actually drawn on the map. Would you prefer DrawnLayerData? RenderedLayerData?

Reply:
Maybe LayerDisplayData? I like that slightly more.